test/helpers/envtest.go

```diff
@@ -130,3 +134,35 @@ func (t *TestEnvironment) Stop() error {
 func (t *TestEnvironment) CreateKubeconfigSecret(cluster *clusterv1.Cluster) error {
 	return kubeconfig.CreateEnvTestSecret(t.Client, t.Config, cluster)
 }
+
+type cleanup func() error
+
+// CreateObj wraps around client.Create and creates the object. It returns a
+// cleanup function that deletes the speicifed resource.
+func (t *TestEnvironment) CreateObj(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) (cleanup, error) {
```

Comment:
@wfernandes I'm still wondering if we can avoid returning a cleanup func.
What if we implement a `testenv.Cleanup` taking in input a list of runtime.Object, so you can do 
```
defer func() {
	g.Expect(testenv.cleanup(namespace, object1, object2)).To(Succeed())
}()			
```

Reply:
Eventually, can also implement `testenv.CleanupNamespace` that lists all the object in a namespace and deletes them

Reply:
@fabriziopandini I updated to your suggestion above. LMK how it looks. Thanks.